### PR TITLE
add cordon and uncordon action to clusternode 

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/13623 
requires - https://github.com/rancher/types/pull/454

### CODE REVIEW

File: pkg/api/customization/node/node.go [hunk 1]
```diff
@@ -31,4 +38,44 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
...
+	var node map[string]interface{}
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &node); err != nil {
+		return httperror.NewAPIError(httperror.InvalidReference, "Error accessing node")
+	}
+	schema := apiContext.Schemas.Schema(&managementschema.Version, client.NodeType)
+	unschedulable := convert.ToBool(values.GetValueN(node, "unschedulable"))
+	if cordon && unschedulable {
+		return httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("Node %s already cordoned", apiContext.ID))
+	} else if !(cordon || unschedulable) {
+		return httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("Node %s already schedulable", apiContext.ID))
```
Comment (alena1108) [inline pkg/api/customization/node/node.go]: it feels like this message can be ambiguous based on cordon/unschedulable values. Can we actually throw some sort of generalized error in situation when `cordon == unschedulable` (it should cover both use cases)?

File: pkg/api/customization/node/node.go [hunk 2]
```diff
@@ -31,4 +38,44 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
...
+	}
+	values.PutValue(node, !unschedulable, "unschedulable")
+	_, err := schema.Store.Update(apiContext, schema, node, apiContext.ID)
```
Comment (alena1108) [inline pkg/api/customization/node/node.go]: can we just return the error? or is there a specific reason for customizing it

File: pkg/controllers/user/nodesyncer/cordonfieldssyncer.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package nodesyncer
+
+import (
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (m *NodesSyncer) syncCordonFields(key string, obj *v3.Node) error {
+	if obj == nil {
```
Comment (alena1108) [inline pkg/controllers/user/nodesyncer/cordonfieldssyncer.go]: || object.DeletionTimestamp != nil

File: pkg/controllers/user/nodesyncer/cordonfieldssyncer.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		return nil
+	}
+	machine := obj.DeepCopy()
```
Comment (alena1108) [inline pkg/controllers/user/nodesyncer/cordonfieldssyncer.go]: why are we doing a deep copy? it should be done only when we update the machine, and we should be scarce and do it only when something gets changed on the update

File: pkg/controllers/user/nodesyncer/cordonfieldssyncer.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	if err != nil {
+		return err
+	}
+	node, err := m.getNode(machine, nodes)
+	if err != nil {
+		return err
+	}
+	shouldUpdate := false
+	if node.Spec.Unschedulable != machine.Spec.InternalNodeSpec.Unschedulable {
+		node.Spec.Unschedulable = machine.Spec.InternalNodeSpec.Unschedulable
```
Comment (alena1108) [inline pkg/controllers/user/nodesyncer/cordonfieldssyncer.go]: we should do deep copy on the node before updating it with the field

File: pkg/api/customization/node/node.go (hunk 2 above)
Comment (kinarashah) [reply pkg/api/customization/node/node.go]: just wanted to distinguish between update by actionName vs controller update

**Comment by alena1108** [comment]
LGTM

**Pull request merged**